src/dbnode/storage/index/field_terms_iterator_test.go

```diff
@@ -99,6 +102,141 @@ func TestFieldsTermsIteratorTermsOnly(t *testing.T) {
 	}, slice)
 }
 
+func TestFieldsTermsIteratorEmptyTerm(t *testing.T) {
+	ctrl := gomock.NewController(xtest.Reporter{t})
+	defer ctrl.Finish()
+
+	seg := newMockSegment(ctrl, map[string][]string{
+		"a": nil,
+	})
+	iter, err := newFieldsAndTermsIterator(seg, fieldsAndTermsIteratorOpts{iterateTerms: false})
+	require.NoError(t, err)
+	slice := toSlice(t, iter)
+	requireSlicesEqual(t, []pair{pair{"a", ""}}, slice)
+}
+
+func TestFieldsTermsIteratorEmptyTermInclude(t *testing.T) {
+	ctrl := gomock.NewController(xtest.Reporter{t})
+	defer ctrl.Finish()
+
+	seg := newMockSegment(ctrl, map[string][]string{
+		"a": nil,
+	})
+	iter, err := newFieldsAndTermsIterator(seg, fieldsAndTermsIteratorOpts{iterateTerms: true})
+	require.NoError(t, err)
+	slice := toSlice(t, iter)
+	requireSlicesEqual(t, []pair{}, slice)
+}
+
```

Comment:
Can you add a few tests with `allowFn` set, maybe one which goes through a couple of segments

Reply:
re: `allowFn` tests - does the existing `TestFieldsTermsIteratorSimpleSkip` not capture stuff you're worried about?

Reply:
re: multiple segments, do you mean Reset()?

Reply:
also, re `allowFn` - the prop tests using `fieldsTermsIteratorPropInput`  are generating test situations with those cases too.

Reply:
My bad yeah, was looking at a partial diff so the tests seemed lacking